rpcs3/rpcs3qt/gs_frame.cpp

```diff
@@ -299,6 +302,96 @@ void gs_frame::flip(draw_context_t, bool /*skip_frame*/)
 	}
 }
 
+void gs_frame::take_screenshot(std::vector<u8> ss_data, const u32 ss_width, const u32 ss_height, bool vulkan_buf)
+{
+	std::thread(
+	    [ss_width, ss_height, vulkan_buf](std::vector<u8> sshot_data) {
+		    std::string screen_path = fmt::format("%s/Screenshots/", fs::get_config_dir());
+
+		    if (!fs::is_dir(screen_path))
+		    {
+			    if (!fs::create_dir(screen_path))
+			    {
+				    LOG_ERROR(GENERAL, "Failed to create screenshot path: %s", screen_path);
+				    return;
+			    }
+		    }
+
+		    u32 file_index;
+		    for (file_index = 0; file_index < 100000000; file_index++)
+		    {
+			    std::string filename = fmt::format("%sScreenshot-%08d.png", screen_path, file_index);
+			    if (fs::is_file(filename))
+				    continue;
+
+			    fs::file ss_file(filename, fs::open_mode::create + fs::open_mode::write);
+			    if (!ss_file)
+			    {
+				    LOG_ERROR(GENERAL, "[Screenshot] Failed to save screenshot to %s", filename);
+				    return;
+			    }
+
+			    std::vector<u8> ss_data_alpha(sshot_data.size());
+			    const u32* ss_ptr = (const u32*)sshot_data.data();
+			    u32* alpha_ptr    = (u32*)ss_data_alpha.data();
+
+			    if (vulkan_buf)
+			    {
+				    for (size_t index = 0; index < sshot_data.size() / sizeof(u32); index++)
```

Comment:
You don't need to do this actually. Just using PNG_TRANSFORM_BGR will swap RGBA to BGRA for you automatically.

Reply:
technically it does the same thing I do though, and in both case I need to set alpha to 255 otherwise sometimes I get a complete transparent screenshot(subway station at the p5 menu for example).

Reply:
There is an option for that too. You don't need to do it manually.